src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MAX_ARRAY_SIZE = 10;
+    private int size;
+    private T[] listsOfElements;
+
+    public ArrayList() {
+        listsOfElements = (T[]) new Object[MAX_ARRAY_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == listsOfElements.length) {
+            grow();
```

Comment:
we could move this check `if (size == listsOfElements.length)` into `grow` method(this will reduce number if duplicated code). If we do this, then it might be worth to rename `grow` method to `ensureCapacity` or smth like this